enhancements/network/on-prem-mutable-vips.md

```diff
@@ -0,0 +1,199 @@
+---
+title: on-prem-mutable-vips
+authors:
+  - "@mkowalski"
+reviewers:
+  - "@JoelSpeed, to review API"
+  - "@sinnykumari, to review MCO"
+  - "@cybertron, to peer-review OPNET"
+  - "@cgwalters"
+approvers:
+  - "@JoelSpeed"
+api-approvers:
+  - "@danwinship"
+  - "@JoelSpeed"
+creation-date: 2023-08-28
+last-updated: 2023-08-28
+tracking-link:
+  - https://issues.redhat.com/browse/OCPSTRAT-178
+  - https://issues.redhat.com/browse/OPNET-340
+  - https://issues.redhat.com/browse/OPNET-80
+see-also:
+  - "/enhancements/on-prem-dual-stack-vips.md"
+replaces:
+superseded-by:
+---
+
+# On-Prem Mutable VIPs
+
+## Summary
+
+Originally the on-prem loadbalancer architecture supported only single-stack IPv4 or IPv6 deployments. Later on, after dual-stack support was added to the on-prem deployments, a work has been done to allow installing clusters with a pair of virtual IPs (one per IP stack). This work however only covered installation-time, leaving clusters originally installed as single-stack and later converted to dual-stack out of scope.
+
+This design is proposing a change which will allow adding a second pair of virtual IPs to clusters that became dual-stack only during their lifetime.
+
+## Motivation
+
+We have customers who installed clusters before dual-stack VIP feature existed. Currently they have no way of migrating because the new feature is avaialble only during initial installation of the cluster.
+
+### User Stories
+
+* As a deployer of a dual-stack OpenShift cluster, I want to access the API using both IPv4 and IPv6.
+
+* As a deployer of a dual-stack OpenShift cluster, I want to access the Ingress using both IPv4 and IPv6.
+
+### Goals
+
+Allow adding a second pair of virtual IPs to an already installed dual-stack cluster.
+
+### Non-Goals
+
+* Modifying existing virtual IP configuration.
+
+* Modifying virtual IP configuration after second pair of VIPs has been added. Right now we only want to cover "create" operation. Rest of CRUD operations will be covered in the future depending on the needs.
```

Comment:
I'm tempted to say we just won't allow changing this. Adding a VIP is one thing, changing one is a completely different kettle of fish and at this point there are no plans to allow that.

Obviously that could change if an important enough use cases is discovered, but I don't want to even suggest that this might happen in the future.

Reply:
Of course, I will rewrite this. I am only concerned about a scenario when someone adds a 2nd pair of VIPs and quickly afterwards they realize they made a typo. But as it's difficult to put a line between "fixing a typo" and "changing completely" I just want to forbid any changes at this stage (and as you suggest, forever)

Reply:
Fixed